go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -932,7 +932,23 @@ func (tsv *TabletServer) ExecuteBatch(ctx context.Context, target *querypb.Targe
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "cannot start a new transaction in the scope of an existing one")
 	}
 
+	if tsv.enableHotRowProtection && asTransaction && len(queries) == 1 {
+		// Serialize transactions which target the same hot row range.
+		// NOTE: We put this intentionally at this place *before* tsv.startRequest()
+		// gets called below. Otherwise, startRequest() from below would cross with
+		// the startRequest() call executed by tsv.beginWaitForSameRangeTransactions().
```

Comment:
do we still need to call startRequest below then, if beginWaitForSameRangeTransactions calls it already?

Reply:
Yes, we do. That's because `beginWaitForSameRangeTransactions` also calls endRequest().

i.e. it looks like this:

```
ExecuteBatch
- beginWaitForSameRangeTransactions
  - startRequest
  - endRequest
- startRequest
- endRequest
```

We have a similar pattern in `BeginExecute()`. There we have actually three of these sections (wait for hot row, Begin(), Execute()).

I've updated the comment in the code to better explain this.